src/test/groovy/org/prebid/server/functional/tests/privacy/AuctionGppSpec.groovy

```diff
@@ -0,0 +1,226 @@
+package org.prebid.server.functional.tests.privacy
+
+import org.prebid.server.functional.model.request.auction.BidRequest
+import org.prebid.server.functional.model.request.auction.Regs
+import org.prebid.server.functional.model.request.auction.User
+import org.prebid.server.functional.model.response.auction.ErrorType
+import org.prebid.server.functional.tests.BaseSpec
+import org.prebid.server.functional.util.PBSUtils
+import org.prebid.server.functional.util.privacy.CcpaConsent
+import org.prebid.server.functional.util.privacy.TcfConsent
+import org.prebid.server.functional.util.privacy.gpp.TcfEuV2Consent
+import org.prebid.server.functional.util.privacy.gpp.UspV1Consent
+
+import static org.prebid.server.functional.model.request.GppSectionId.TCF_EU_V2
+import static org.prebid.server.functional.model.request.GppSectionId.USP_V1
+import static org.prebid.server.functional.util.privacy.CcpaConsent.Signal.ENFORCED
+import static org.prebid.server.functional.util.privacy.CcpaConsent.Signal.NOT_ENFORCED
+import static org.prebid.server.functional.util.privacy.TcfConsent.GENERIC_VENDOR_ID
+import static org.prebid.server.functional.util.privacy.TcfConsent.PurposeId.BASIC_ADS
+
+class AuctionGppSpec extends BaseSpec {
+
+    def "PBS should populate gdpr to 1 when regs.gdpr is not specified and gppSid contains 2"() {
+        given: "Default bid request with gppSid and without gdpr"
+        def gppSidsId = [TCF_EU_V2.valueAsInt]
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs = new Regs(gdpr: null, gppSid: gppSidsId)
+        }
+
+        when: "PBS processes auction request"
+        def bidResponse = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain regs.gdpr and gppSid from request"
```

Comment:
This is not bidder request. Why are you checking resolved request instead of the actual bidder call in the first place? Or bidder request doesn't happen because we lack the consent string? If so, please add a check that bidder wasn't called.